tokio-trace/src/span.rs

```diff
@@ -64,10 +200,9 @@
 //!     type Error = ();
 //!
 //!     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-//!         self.span.in_scope(|| {
-//!             // Do actual future work
+//!         let _enter = self.span.enter();
```

Comment:
If this is a Mutex-like scopeguard, why assign to `_enter`?

Reply:
For some reason, I had the wrong understanding of how MutexGuards worked. Feel free to disregard.